opentelemetry/proto/trace/v1/trace.proto

```diff
@@ -329,5 +343,26 @@ enum SpanFlags {
   // Bits 0-7 are used for trace flags.
   SPAN_FLAGS_TRACE_FLAGS_MASK = 0x000000FF;
 
-  // Bits 8-31 are reserved for future use.
+  // Bits 8 and 9 are used to indicate that the parent span or link span is remote.
+  // See SpanFlagsContextIsRemote for the valid values.
+  SPAN_FLAGS_CONTEXT_IS_REMOTE_MASK = 0x00000300;
+
+  // Bits 10-31 are reserved for future use.
+}
+
+// SpanFlagsContextIsRemote represents constants used to interpret
+// bits 8 and 9 of the Spans.flags or Link.flags field. These two bits
+// represent whether the parent or link is remote. Bit 8 represents whether
+// the field has been set and bit 9 represents whether the parent or link is
+// remote.
+// Once the bits have been read using SPAN_FLAGS_CONTEXT_IS_REMOTE_MASK,
+// the following constants can be used to interpret them.
+enum SpanFlagsContextIsRemote {
+  // Older client not supporting this flag, or not known whether parent or
+  // linked span is remote: 00
+  SPAN_FLAGS_CONTEXT_IS_REMOTE_UNKNOWN = 0;
+  // The parent or linked span is remote: 11
+  SPAN_FLAGS_CONTEXT_IS_REMOTE = 0x00000300;
+  // The parent or linked span is not remote: 01
+  SPAN_FLAGS_CONTEXT_IS_NOT_REMOTE = 0x00000100;
```

Comment:
Bit random choice, why not `00` - unknown, `01` - false, `10` - true?

Reply:
@bogdandrutu The idea was that the first bit indicates whether the value is set or not and the second bit represents `false`/`true`:

- `00` unknown
- `x1` 'known' 
  - `01` false
  - `11` true

> 00 - unknown, 01 - false, 10 - true

What would be the logic with that? How would that be better?

Does it really matter at the end? As long as we have three different values that follow _some_ logic it should be fine, right?

Reply:
It may or may not be better, but feels more natural to have the first 3 values used. I agree that is fine, was curios about the logic behind, so just document this and it is fine.

Or actually based on this, you can have 2 bits, first that says if the second is "known". Then I feel that `00`, `10` (false) and `11`(true) makes more sense, and we can have 2 different masks `SPAN_FLAGS_CONTEXT_HAS_IS_REMOTE_MASK` and `SPAN_FLAGS_CONTEXT_IS_REMOTE_MASK`

Reply:
@bogdandrutu see this comment https://github.com/open-telemetry/opentelemetry-proto/pull/484#discussion_r1329036599

Reply:
@bogdandrutu I think I agree. If we simply call this a 2-bit field, then we can actually use 4 values.

`00` - unknown legacy
`01` - context is not remote
`10` - context is remote
`11` - unknown, not legacy

I.e., we can distinguish unknown-remoteness from legacy-unspecified-remoteness.

For a consumer of this data inside the collector, say, I would expect to see first-class accessor methods anyway, e.g., `ContextRemoteness() RemotenessState` with no bit-fields to speak of, just the 4 potential states the two bits can have.

Reply:
>For a consumer of this data inside the collector, say, I would expect to see first-class accessor methods anyway, e.g., ContextRemoteness() RemotenessState with no bit-fields to speak of, just the 4 potential states the two bits can have.

We will definitely do that and hide all the complexity to the user.

Reply:
> unknown, not legacy

What would that mean? Do you propose to make isRemote a tristate in the OTel data model?

Reply:
@Oberon00 I was suggesting that it could be done in the future, if someone could find a reason. Do we always know if a context is or is not remote? Either way, I would like to unblock this PR and I am actually happy with it either way. I do not see a significant advantage either way, but if we're forced to look for pros/cons, this was all I could find.